src/buildstream/storage/_casbaseddirectory.py

```diff
@@ -529,6 +529,10 @@ def is_dir_in(entry: _IndexEntry, directory: CasBasedDirectory) -> bool:
 
         self.__invalidate_digest()
 
+    def get_cas_digest(self):
```

Comment:
This adds a public method to a private class, and it feels weird.

If we want to add a public method, it should be added to the base `Directory` class and be implemented by both subclasses.

That said, I feel we should not add public method at this time as nothing is to be exposed to plugins. We can consider adding the public method later in #1991 and just use the internal methods for now.